### PR TITLE
Feat/emails

### PR DESCRIPTION
**Pull request by rafinskipg**

### CODE REVIEW

File: components/email/email-form/index.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  user: UserType
+}
+
+export default function EmailForm({
+  onSubmitted,
+  onCancel,
+  user
+}: PropTypes) {
+  const [from, setFrom] = useState(user ? user.email : '')
+  const [to, setTo] = useState('')
```
Comment (ihorkitrum) [inline components/email/email-form/index.tsx:21]: maybe it's better to use useReducer hook?

File: components/email/email-form/index.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(false)
+  const [errorEmptyFields, setErrorEmptyFields] = useState(false)
+
+  useEffect(() => {
+    setFrom(user ? user.email : '')
+  }, [user])
+
+  const onSubmit = (ev: React.FormEvent<HTMLFormElement>) => {
```
Comment (ihorkitrum) [inline components/email/email-form/index.tsx:35]: it's better to wrap functions with the useCallback hook

File: components/email/email-form/index.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  }
+
+  return (
+    <>
+      <div className="email-form">
+        <h3>New email</h3>
+        <form onSubmit={onSubmit}>
+        <DynamicFieldEdit
+            value={from}
+            onChange={(val) => setFrom(val)}
```
Comment (ihorkitrum) [inline components/email/email-form/index.tsx:87]: it's the same as `onChange={setFrom}`

File: components/email/email-form/index.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    ev.stopPropagation()
+    onCancel()
+  }
+
+  return (
+    <>
+      <div className="email-form">
+        <h3>New email</h3>
+        <form onSubmit={onSubmit}>
+        <DynamicFieldEdit
```
Comment (ihorkitrum) [inline components/email/email-form/index.tsx:85]: maybe it's better to define config with fields and then use the loop to call DynamicFieldEdit component. like it was done [there](https://github.com/nucleo-org/edge-next/blob/master/components/user/admin-user/new/user-form.tsx)

File: components/email/email-table/index.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    url: `${API.email}`,
+    limit,
+    sortBy,
+    sortOrder,
+    config: {
+      revalidateOnFocus: false,
+    },
+  })
+
+  const changeOrdering = useCallback(
```
Comment (ihorkitrum) [inline components/email/email-table/index.tsx:40]: I think we should create hook with such logic

File: components/generic/emails-input/index.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+import { memo, useEffect, useState } from 'react'
+
+function Email(props) {
+  const valid = EmailValidator.validate(props.email)
+  return (
+    <>
+      <div className={`email ${!valid ? 'error' : ''}`}>
+        {props.email}{' '}
+        <span className="close" onClick={props.onClickRemove}>
+          <svg
```
Comment (ihorkitrum) [inline components/generic/emails-input/index.tsx:12]: it's better not to use svg code directly in the component. it's better to create .svg file and import it whenever it needs

File: components/generic/emails-input/index.tsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+        #close-path {
+          fill: var(--edge-foreground);
+        }
+      `}</style>
+    </>
+  )
+}
+
+function EmailsInput(props) {
```
Comment (ihorkitrum) [inline components/generic/emails-input/index.tsx:63]: it's a good practice to create the only one component per file

File: components/email/email-form/index.tsx (hunk 4 above)
Comment (rafinskipg) [reply components/email/email-form/index.tsx:85]: no its more difficult to style

**Pull request merged**